scylla/src/transport/query_result.rs

```diff
@@ -0,0 +1,719 @@
+use bytes::Bytes;
+use thiserror::Error;
+use uuid::Uuid;
+
+use scylla_cql::frame::frame_errors::ParseError;
+use scylla_cql::frame::response::result::{ColumnSpec, ColumnType, RawRows, Row, TableSpec};
+use scylla_cql::types::deserialize::result::TypedRowIterator;
+use scylla_cql::types::deserialize::row::DeserializeRow;
+use scylla_cql::types::deserialize::{DeserializationError, TypeCheckError};
+
+use super::legacy_query_result::LegacyQueryResult;
+
+/// A view over specification of a table in the database.
+#[derive(Debug, Clone, Copy)]
+#[cfg_attr(test, derive(PartialEq, Eq))]
+pub struct TableSpecView<'res> {
+    table_name: &'res str,
+    ks_name: &'res str,
+}
```

Comment:
As `TableSpec` is defined in `scylla-cql`, it must have public API (a public constructor and/or public fields), for `scylla` crate to be able to access it. Hence, we don't want to expose `TableSpec` to users at all; instead, it's better to only expose types defined in `scylla`, because then we can truly control visibility using `pub` and `pub(crate)`. 

This is a pattern that I believe we should employ commonly, because then we will be able to make code adjustments to inner types, even changing their names and/or types, without introducing breaking API changes.

Reply:
Is a public contructor a problem in this case? I don't see why, but probably I don't understand something.

This patterns means basically duplicating a lot of code from scylla-cql. If we have to do this then maybe scylla-cql should not exist?

Reply:
1. scylla-cql was created specifically to allow code reuse for scylla-proxy. As long as scylla-proxy exists and we don't want to duplicate low-level ser/de code there, we need scylla-cql.
2. You see this as duplicating code. I see this as introducing a new layer - stable abstraction. This is something that a stable driver should have, independently of its inner working. I have an impression that Rust driver has not yet developed such layer, and its API more-or-less reflects its inner implementation details. For the sake of future compatibility and more freedom to modify things after 1.0, I believe we need such layer, consisting of exposed yet extremely stable types. Even if in some cases it boils down to an identity-or-nearly-identity layer.

Reply:
To be honest, I hate the dual world of scylla-cql and scylla. I feel unsafe about those `pub`s in scylla-cql.
Perhaps we could at least move some part of scylla-cql to scylla. It seems that scylla-proxy does not use the serialization nor deserialization framework anyway (even though I initially intended it to do so - to be capable of e.g. injecting arbitrary data).